### PR TITLE
Deprecate internal packages

### PR DESCRIPTION
**Pull request by grouma**
Mark internal packages as deprecated to hopefully prevent accidental use.

### CODE REVIEW

File: pkgs/test_core/pubspec.yaml [hunk 1]
```diff
@@ -32,4 +32,4 @@ dependencies:
   # properly constrains all features it provides.
   matcher: ">=0.12.5 <0.12.6"
   # Use an exact version until the test_api package is stable.
-  test_api: 0.2.6
+  test_api: 0.2.7
```
Comment (jakemac53) [inline pkgs/test_core/pubspec.yaml]: does this also need a dep override? or just going to submit on red?

Comment (grouma) [reply pkgs/test_core/pubspec.yaml]: This was an oversight.

File: pkgs/test_api/lib/backend.dart [hunk 2]
```diff
@@ -2,6 +2,10 @@
 // for details. All rights reserved. Use of this source code is governed by a
 // BSD-style license that can be found in the LICENSE file.
 
+@Deprecated('package:test_api is not intended for general use. '
```
Comment (natebosch) [inline pkgs/test_api/lib/backend.dart]: Do we see this one coming up in practice?

File: pkgs/test_core/lib/backend.dart [hunk 3]
```diff
@@ -2,6 +2,11 @@
 // for details. All rights reserved. Use of this source code is governed by a
 // BSD-style license that can be found in the LICENSE file.
 
+@Deprecated('package:test_core is not intended for general use. '
+    'Please use package:test.')
+library test_core;
```
Comment (natebosch) [inline pkgs/test_core/lib/backend.dart]: `test_core.backend`

File: pkgs/test_api/lib/backend.dart [hunk 4]
```diff
@@ -2,6 +2,10 @@
...
+    'Please use package:test.')
+library test_api;
```
Comment (natebosch) [inline pkgs/test_api/lib/backend.dart]: `test_api.backend`.

I don't know if there is a requirement for unique library names - but we should aim for it.

File: pkgs/test_api/lib/backend.dart (hunk 2 above)
Comment (grouma) [reply pkgs/test_api/lib/backend.dart]: Yes. I can't find it but there was a flutter bug recently where folks were importing from test_api and it caused unexpected issues.

File: pkgs/test_api/lib/backend.dart (hunk 4 above)
Comment (grouma) [reply pkgs/test_api/lib/backend.dart]: Yeah I wasn't sure if having the same library name would be problematic but it didn't appear to cause issues. Updated it nonetheless.

**Pull request merged**